src/systems/rf_comms/RFComms.cc

```diff
@@ -0,0 +1,464 @@
+struct RangeConfiguration
+{
+  /// \brief Hard limit on range.
+  double maxRange = 50.0;
+
+  /// \brief Fading exponent.
+  double fadingExponent = 2.5;
+
+  /// \brief Received power at 1m (in dBm).
+  double l0 = 40;
+
+  /// \beief Standard deviation for received power.
+  double sigma = 10;
+
+  /// Output stream operator.
+  /// @param[out] _oss Stream.
+  /// @param[in] _config configuration to output.
+  friend std::ostream &operator<<(std::ostream &_oss,
+                                  const RangeConfiguration &_config)
+  {
+    _oss << "RF Configuration (range-based)" << std::endl
+         << "-- max_range: " << _config.maxRange << std::endl
+         << "-- fading_exponent: " << _config.fadingExponent << std::endl
+         << "-- l0: " << _config.l0 << std::endl
+         << "-- sigma: " << _config.sigma << std::endl;
+
+    return _oss;
+  }
+};
+
+/// \brief Radio configuration parameters.
+///
+/// Static parameters such as channel capacity and transmit power.
+struct RadioConfiguration
+{
+  /// \brief Capacity of radio in bits-per-second.
+  double capacity = 54000000;
+
+  /// \brief Default transmit power in dBm. Default is 27dBm or 500mW.
+  double txPower = 27;
+
+  /// \brief Modulation scheme, e.g., QPSK (Quadrature Phase Shift Keyring).
+  std::string modulation = "QPSK";
+
+  /// \brief Noise floor of the radio in dBm.
+  double noiseFloor = -90;
+
+  /// Output stream operator.
+  /// @param _oss Stream.
+  /// @param _config configuration to output.
+  friend std::ostream &operator<<(std::ostream &_oss,
+                                  const RadioConfiguration &_config)
+  {
+    _oss << "Radio Configuration" << std::endl
+         << "-- capacity: " << _config.capacity << std::endl
+         << "-- tx_power: " << _config.txPower << std::endl
+         << "-- noise_floor: " << _config.noiseFloor << std::endl
+         << "-- modulation: " << _config.modulation << std::endl;
+
+    return _oss;
+  }
+};
+
+/// \brief Store radio state
+///
+/// Structure to hold radio state including the pose and book-keeping
+/// necessary to implement bitrate limits.
+struct RadioState
+{
+  /// \brief Timestamp of last update.
+  double timeStamp;
+
+  /// \brief Pose of the radio.
+  ignition::math::Pose3<double> pose;
+
+  /// \brief Recent sent packet history.
+  std::list<std::pair<double, uint64_t>> bytesSent;
+
+  /// \brief Accumulation of bytes sent in an epoch.
+  uint64_t bytesSentThisEpoch = 0;
+
+  /// \brief Recent received packet history.
+  std::list<std::pair<double, uint64_t>> bytesReceived;
+
+  /// \brief Accumulation of bytes received in an epoch.
+  uint64_t bytesReceivedThisEpoch = 0;
+};
+
+/// \brief Type for holding RF power as a Normally distributed random variable.
+struct RFPower
+{
+  /// \brief Expected value of RF power.
+  double mean;
+
+  /// \brief Variance of RF power.
+  double variance;
+
+  /// \brief double operator.
+  /// \return the RFPower as a double.
+  operator double()
```

Comment:
```suggestion
  operator double() const
```

Reply:
Changed in [991b690](https://github.com/ignitionrobotics/ign-gazebo/pull/1428/commits/991b6906a169f4d63bbcdf5b4bd72ace57848d6b)